### PR TITLE
feat(azure): add new check related with Public IPs in Shodan.io

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

This pr adds a new check:

* network_public_ip_shodan

and fixes typo in all tests that don´t include azure_fixtures

### Description

This check ensures that no public ip addresses are shown in shodan

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/network/network_service.py [hunk 1]
```diff
@@ -10,9 +10,11 @@
 class Network(AzureService):
     def __init__(self, audit_info):
         super().__init__(NetworkManagementClient, audit_info)
+        self.audit_info = audit_info
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/network/network_service.py]:
```suggestion
```

File: prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+from prowler.lib.check.models import Check, Check_Report_Azure
+from prowler.lib.logger import logger
+from prowler.providers.azure.services.network.network_client import network_client
+
+
+class network_public_ip_shodan(Check):
+    def execute(self):
+        findings = []
+        shodan_api_key = network_client.audit_info.audit_config.get("shodan_api_key")
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py]:
```suggestion
        shodan_api_key = network_client.audit_config.get("shodan_api_key")
```

File: prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                    try:
+                        shodan_info = api.host(ip.ip_address)
+                        report.status = "FAIL"
+                        report.status_extended = f"Public IP {ip.ip_address} listed in Shodan with open ports {str(shodan_info['ports'])} and ISP {shodan_info['isp']} in {shodan_info['country_name']}. More info at https://www.shodan.io/host/{ip.ip_address}."
+                        findings.append(report)
+                    except shodan.APIError as error:
+                        if "No information available for that IP" in error.value:
+                            report.status = "PASS"
+                            report.status_extended = (
+                                f"Elastic IP {ip.ip_address} is not listed in Shodan."
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py]:
```suggestion
                                f"Public IP {ip.ip_address} is not listed in Shodan."
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments.

File: prowler/providers/azure/services/network/network_service.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/azure/services/network/network_service.py]: Done

File: prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py]: Done

File: prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/azure/services/network/network_public_ip_shodan/network_public_ip_shodan.py]: Done

**Comment by pedrooot** [comment]
No merge until https://github.com/prowler-cloud/prowler/pull/3432 is merged

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**